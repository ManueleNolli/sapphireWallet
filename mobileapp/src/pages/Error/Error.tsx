import { Layout, Text } from '@ui-kitten/components'
import { appStyles, vw } from '../../Styles'
import React from 'react'
import { Image } from 'expo-image'
import { attention } from '../../assets/AssetsRegistry'
import { StyleSheet } from 'react-native'

type ErrorProps = {
  text?: string
}
export default function Error({ text }: ErrorProps) {
  return (
    <Layout style={appStyles.center}>
      <Image style={styles.image} source={attention} />
      {text && <Text category="h6">{text}</Text>}
    </Layout>
  )
}

const styles = StyleSheet.create({
  image: {
    width: 50 * vw,
    height: 50 * vw,
    maxWidth: 250,
    maxHeight: 250,
  },
})
