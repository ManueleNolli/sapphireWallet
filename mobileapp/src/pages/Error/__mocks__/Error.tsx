const ErrorMock = () => <div data-testid={'Error'}>Error</div>

export default ErrorMock
