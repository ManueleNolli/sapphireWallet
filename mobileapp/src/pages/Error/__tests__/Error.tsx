import renderWithTheme from '../../../TestHelper'
import React from 'react'
import Error from '../Error'

describe('Error', () => {
  it('Render correctly without text', () => {
    const tree = renderWithTheme(<Error />)

    expect(tree).toMatchSnapshot()
  })

  it('Render correctly with text', () => {
    const tree = renderWithTheme(<Error text="Error!" />)

    expect(tree).toMatchSnapshot()
  })
})
