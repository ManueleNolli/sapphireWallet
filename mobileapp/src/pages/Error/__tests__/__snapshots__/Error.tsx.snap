// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Render correctly with text 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={375}
    maxHeight={250}
    maxWidth={250}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={[]}
    style={
      {
        "height": 375,
        "maxHeight": 250,
        "maxWidth": 250,
        "width": 375,
      }
    }
    transition={null}
    width={375}
  />
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        undefined,
      ]
    }
  >
    Error!
  </Text>
</View>
`;

exports[`Error Render correctly without text 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={375}
    maxHeight={250}
    maxWidth={250}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={[]}
    style={
      {
        "height": 375,
        "maxHeight": 250,
        "maxWidth": 250,
        "width": 375,
      }
    }
    transition={null}
    width={375}
  />
</View>
`;
